web/main/models.py

```diff
@@ -1060,15 +1057,15 @@ def casebook_fts(
         Search in casebook by query:
         >>> assert dump_search_results([FullTextSearchIndex().casebook_fts(casebooks[0].id, "legal_doc_fulltext", query_str='Dubious'),]) == (
         ...     [
-        ...         {'citations': 'Adventures in criminality, 1 Fake 1, (2001)', 'display_name': 'Legal Doc 0', 'jurisdiction': None, 'effective_date': '1900-01-01T00:00:00+00:00', 'effective_date_formatted': 'January   1, 1900', 'headlines': ['<b>Dubious</b> legal claim']},
-        ...         {'citations': 'Adventures in criminality, 1 Fake 1, (2001)', 'display_name': 'Legal Doc 1', 'jurisdiction': None, 'effective_date': '1900-01-01T00:00:00+00:00', 'effective_date_formatted': 'January   1, 1900', 'headlines': ['<b>Dubious</b> legal claim']},
-        ...         {'citations': 'Adventures in criminality, 1 Fake 1, (2001)', 'display_name': 'Legal Doc 2', 'jurisdiction': None, 'effective_date': '1900-01-01T00:00:00+00:00', 'effective_date_formatted': 'January   1, 1900', 'headlines': ['<b>Dubious</b> legal claim']}
+        ...         {'citations': 'Adventures in criminality, 1 Fake 1, (2001)', 'display_name': 'Legal Doc 0', 'jurisdiction': None, 'effective_date': '1900-01-01T00:00:00+00:00', 'effective_date_formatted': 'January   1, 1900', 'headlines': ['<b>Dubious</b> legal claim'], 'ordinals': ''},
+        ...         {'citations': 'Adventures in criminality, 1 Fake 1, (2001)', 'display_name': 'Legal Doc 1', 'jurisdiction': None, 'effective_date': '1900-01-01T00:00:00+00:00', 'effective_date_formatted': 'January   1, 1900', 'headlines': ['<b>Dubious</b> legal claim'], 'ordinals': ''},
+        ...         {'citations': 'Adventures in criminality, 1 Fake 1, (2001)', 'display_name': 'Legal Doc 2', 'jurisdiction': None, 'effective_date': '1900-01-01T00:00:00+00:00', 'effective_date_formatted': 'January   1, 1900', 'headlines': ['<b>Dubious</b> legal claim'], 'ordinals': ''}
```

Comment:
The smallest of nitpicks: I find it a little confusing that these text docs are nearly identical to each other. Is that helpful for simplicity, making it clear which fields do and do not matter, etc., or would these tests be just as effective if it were more obvious that the documents can be arbitrarily different from each other?

Reply:
The appearance of the docs is a consequence of the factory. Honestly, I think these fields don't matter at all --- only the number of results and the highlighting. They've also been a total pain to keep updated as the output of the function changes in tiny tiny ways. I think it makes a lot of sense to check the function output once (all of the fields are there), and then just make sure we get the right number of results in the other cases.

Thanks for pointing this out, let me tweak these tests.

Reply:
I'll make a pitch here for a heuristic that if the test is more than just a few lines long, it's probably easier to implement as a standalone `pytest` function rather than a doctest.

Reply:
yes, I changed them to just ensure that the right number of search results popped up, rather than the full content of the searches!

Reply:
That also makes sense, some of these doctests are so long that they interfere with trying to read the function itself (in my opinion)

Reply:
Yeah, TBD with @jcushman: I believe this style was intended to be somewhat of an experiment in [literate programming](https://en.wikipedia.org/wiki/Literate_programming), and an experiment in "developers will read the tests more and write more tests if everything is all together in one file"

Reply:
When we set this up we were definitely playing with the upper boundary on doctests -- in particular asking, can the doctests be written compactly enough that they serve the purpose of a docstring but better, convincing you that a function does what it says it does and teaching you how to call it, without being _too_ long? I still think there's some appeal to that, especially with functions that take and return complex data structures where it's hard to write documentation without showing examples anyway.

But! This is totally a matter of taste and I won't be working on the codebase, so I don't want to have an oar in as far as where to draw the line on length and move to a separate pytest function. Carry on!